File: TravellerCreator/ClientApp/src/app/sector-creator/models/basic/basic-sector.ts
import {ISector} from "../interfaces/sector";
import {BasicSubsector} from "./basic-subsector";
import {SectorType} from "../enums/sector-type";

export class BasicSector implements ISector {
  sectorType: SectorType = SectorType.Basic;
  subsectors: BasicSubsector[] = [];
}

File: TravellerCreatorModels/Basic/BasicSector.cs
using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.Basic;

public class BasicSector : ISector
{
    public SectorType SectorType => SectorType.Basic;
    public List<ISubsector> Subsectors { get; set; } = new();
}